### PR TITLE
Added ElytraFix feature in NCPTweaks

### PR DESCRIPTION
**Pull request by cattyngmd**
very useful feature for 2b2t and other strict servers.

### CODE REVIEW

File: src/main/java/me/earth/earthhack/impl/modules/player/ncptweaks/ListenerPosLook.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import net.minecraft.entity.item.EntityFireworkRocket;
+import net.minecraft.init.Items;
+import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.ItemElytra;
+import net.minecraft.item.ItemStack;
+import net.minecraft.network.play.server.SPacketPlayerPosLook;
+
+final class ListenerPosLook extends ModuleListener<NCPTweaks, PacketEvent.Receive<SPacketPlayerPosLook>> {
+    public ListenerPosLook(NCPTweaks module) {
+        super(module, PacketEvent.Receive.class);
```
Comment (3arthqu4ke) [inline src/main/java/me/earth/earthhack/impl/modules/player/ncptweaks/ListenerPosLook.java:15]: You need to declare the type of the Packet here too, so:
```java
super(module, PacketEvent.Receive.class, SPacketPlayerPosLook.class);
```

File: src/main/java/me/earth/earthhack/impl/modules/player/ncptweaks/ListenerPosLook.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    @Override
+    public void invoke(PacketEvent.Receive<SPacketPlayerPosLook> event) {
+        if (!module.elytraFix.getValue()) {
+            return;
+        }
+
+        ItemStack stack = mc.player.getItemStackFromSlot(EntityEquipmentSlot.CHEST);
+
+        if (stack.getItem() == Items.ELYTRA && ItemElytra.isUsable(stack) && mc.player.isElytraFlying()) {
+            for (EntityFireworkRocket e : mc.world.getEntities(EntityFireworkRocket.class, e -> ((IEntityFireworkRocket)e).getBoostedEntity() == mc.player)) {
```
Comment (3arthqu4ke) [inline src/main/java/me/earth/earthhack/impl/modules/player/ncptweaks/ListenerPosLook.java:27]: PacketEvent.Receive gets called on the NetworkThread, not on the main thread, making all of this unsafe. Can you wrap all of this with `mc.addScheduledTask`, inside a null check for mc.player and the world?

**Review by 3arthqu4ke — CHANGES_REQUESTED** [review]

**Comment by 3arthqu4ke** [comment]
I btw saw your [void-esp-cache](https://github.com/cattyngmd/3arthh4ck/tree/voidesp-cache) branch. I think my TODO was a bit missleading. I did not mean to use sphere, because right now you are going to skip over thousands of blocks with the `y != 0` check, but something like a disk specifically made for it. But in hindsight the Sphere cache was kind of a bad idea. My idea was to cache all the sphere offsets sorted by distance to the middle, but honestly a sort operation takes almost no time compared to the checks done in the for loop. 

The only optimization I see there is using a MutableBlockPos instead of allocating a BlockPos for every offest and even with thatI am not sure if it is really necessary.

**Comment by cattyngmd** [comment]
> I btw saw your [void-esp-cache](https://github.com/cattyngmd/3arthh4ck/tree/voidesp-cache) branch. I think my TODO was a bit missleading. I did not mean to use sphere, because right now you are going to skip over thousands of blocks with the `y != 0` check, but something like a disk specifically made for it. But in hindsight the Sphere cache was kind of a bad idea. My idea was to cache all the sphere offsets sorted by distance to the middle, but honestly a sort operation takes almost no time compared to the checks done in the for loop.

yeah, ive tried to make my own realization of AbstractGeoSphere that makes a circle instead of sphere. The benchmark results were almost the same as the current for loop (https://github.com/3arthqu4ke/3arthh4ck/blob/1a6e815ab5b7b3ef59092c9e13516db48abfb89d/src/main/java/me/earth/earthhack/impl/modules/render/voidesp/VoidESP.java#L78-L98).
152300ns avg without the cache.
158840ns avg with the cache.
 
> The only optimization I see there is using a MutableBlockPos instead of allocating a BlockPos for every offest and even with thatI am not sure if it is really necessary.

yep thats right

File: src/main/java/me/earth/earthhack/impl/modules/player/ncptweaks/ListenerPosLook.java [hunk 3]
```diff
@@ -12,22 +12,28 @@
 
 final class ListenerPosLook extends ModuleListener<NCPTweaks, PacketEvent.Receive<SPacketPlayerPosLook>> {
     public ListenerPosLook(NCPTweaks module) {
-        super(module, PacketEvent.Receive.class);
+        super(module, PacketEvent.Receive.class, SPacketPlayerPosLook.class);
     }
 
     @Override
     public void invoke(PacketEvent.Receive<SPacketPlayerPosLook> event) {
+        if (mc.player == null || mc.world == null) {
```
Comment (3arthqu4ke) [inline src/main/java/me/earth/earthhack/impl/modules/player/ncptweaks/ListenerPosLook.java:20]: Sorry, I meant putting this check inside the scheduled task. There is a small chance that between this event running on the packet thread and the scheduled task running on the main thread world or player become null, which will crash the game when the scheduled task runs.

**Review by 3arthqu4ke — CHANGES_REQUESTED** [review]

File: src/main/java/me/earth/earthhack/impl/modules/player/ncptweaks/ListenerPosLook.java (hunk 3 above)
Comment (cattyngmd) [reply src/main/java/me/earth/earthhack/impl/modules/player/ncptweaks/ListenerPosLook.java:20]: ah ok i didnt know that, thanks

**Review by 3arthqu4ke — APPROVED** [review]

**Comment by 3arthqu4ke** [comment]
🔥

**Pull request merged**